pyproject.toml

```diff
@@ -29,10 +29,14 @@ ruff = ">=0.2.0"
 mypy = ">=1.5"
 pytest-cov = ">=3"
 pytest-cookies = ">=0.6.1"
-Sphinx = ">=4.4"
-jupyterlab = ">=3.5.1"
-jupyter-book = ">=0.12.3"
-myst-parser = ">=0.15"
+markdown-callouts = ">=0.2"
+mkdocs = ">=1.3.1"
+mkdocs-include-exclude-files = ">=0.0.1"
+mkdocs-jupyter = ">=0.21"
+mkdocs-material = ">=8.3.9"
+mkdocstrings-python = ">=0.7.1"
+mkdocstrings-python-legacy = ">=0.2.3"
+pymdown-extensions = ">=9.5"
```

Comment:
The full error mentions -
```
Cannot install paginate.
```

Maybe we should add that as a dependency here?

Reply:
I'll try doing it

Reply:
still the same issue, should I try adding `distlib` to dependency?

Reply:
still failing after adding both `paginate` and `distlib` to dependency, yikes

Reply:
The tests were passing before the update of smoke tests in #226. Perhaps this update is causing the error?

Reply:
Maybe @xmnlab can help with this 👀

Reply:
thanks for the ping here @Saransh-cpp 
I managed to resolve that adding paginate to the conda environment directly:

https://github.com/arxlang/astx/blob/main/conda/dev.yaml#L13

this is a very ugly workaround .. but it works
in other projects I needed to add more deps to the conda environment in order to fix this issue with distlib

Reply:
I had this problem long time ago ... and it is still a problem ... tbh I don't know why it is happening or each piece of this pipeline (conda, poetry, etc) is the responsible for the issue

Reply:
Should I use this solution and then remove paginate and distlib from pyproject.toml to make it cleaner?

Reply:
yes I think that it is the way to go

Reply:
Well, I wanted to rebase the branch, but accidentally misclicked on the 'Update Branch' button, and now the tests seem to be passing. I don't understand how, though.

Reply:
yep .. not sure .. but anyway I think you should already that .. because maybe it will pop up later